test/extensions/filters/http/compressor/compressor_filter_speed_test.cc

```diff
@@ -192,102 +264,263 @@ compressChunks8192/5/manual_time        15.9 ms         16.1 ms           45
+static void compressChunks16384WithGzip(benchmark::State& state) {
+  NiceMock<Http::MockStreamDecoderFilterCallbacks> decoder_callbacks;
+  const auto idx = state.range(0);
+  const auto& params = gzip_compression_params[idx];
+
+  for (auto _ : state) { // NOLINT
+    std::vector<Buffer::OwnedImpl> chunks = generateChunks(7, 16384);
+    compressWith(CompressorLibs::Gzip, std::move(chunks), params, decoder_callbacks, state);
+  }
+}
+BENCHMARK(compressChunks16384WithGzip)
+    ->DenseRange(0, 8, 1)
+    ->UseManualTime()
+    ->Unit(benchmark::kMillisecond);
+
+// NOLINTNEXTLINE(readability-identifier-naming)
+static void compressChunks8192WithGzip(benchmark::State& state) {
+  NiceMock<Http::MockStreamDecoderFilterCallbacks> decoder_callbacks;
+  const auto idx = state.range(0);
+  const auto& params = gzip_compression_params[idx];
+
+  for (auto _ : state) { // NOLINT
+    std::vector<Buffer::OwnedImpl> chunks = generateChunks(15, 8192);
+    compressWith(CompressorLibs::Gzip, std::move(chunks), params, decoder_callbacks, state);
+  }
+}
+BENCHMARK(compressChunks8192WithGzip)
+    ->DenseRange(0, 8, 1)
+    ->UseManualTime()
+    ->Unit(benchmark::kMillisecond);
+
+// NOLINTNEXTLINE(readability-identifier-naming)
+static void compressChunks4096WithGzip(benchmark::State& state) {
+  NiceMock<Http::MockStreamDecoderFilterCallbacks> decoder_callbacks;
+  const auto idx = state.range(0);
+  const auto& params = gzip_compression_params[idx];
+
+  for (auto _ : state) { // NOLINT
+    std::vector<Buffer::OwnedImpl> chunks = generateChunks(30, 4096);
+    compressWith(CompressorLibs::Gzip, std::move(chunks), params, decoder_callbacks, state);
+  }
+}
+BENCHMARK(compressChunks4096WithGzip)
+    ->DenseRange(0, 8, 1)
+    ->UseManualTime()
+    ->Unit(benchmark::kMillisecond);
+
+// NOLINTNEXTLINE(readability-identifier-naming)
+static void compressChunks1024WithGzip(benchmark::State& state) {
+  NiceMock<Http::MockStreamDecoderFilterCallbacks> decoder_callbacks;
+  const auto idx = state.range(0);
+  const auto& params = gzip_compression_params[idx];
+
+  for (auto _ : state) { // NOLINT
+    std::vector<Buffer::OwnedImpl> chunks = generateChunks(120, 1024);
+    compressWith(CompressorLibs::Gzip, std::move(chunks), params, decoder_callbacks, state);
+  }
+}
+BENCHMARK(compressChunks1024WithGzip)
+    ->DenseRange(0, 8, 1)
+    ->UseManualTime()
+    ->Unit(benchmark::kMillisecond);
+
+static std::vector<CompressionParams> zstd_compression_params = {
+    // level1 + default
+    {1, 0, 0, 0},
+
+    // level2 + default
+    {2, 0, 0, 0},
+
+    // level3 + default
+    {3, 0, 0, 0},
+
+    // level4 + default
+    {4, 0, 0, 0},
+
+    // level5 + default
+    {5, 0, 0, 0},
+
+    // level6 + default
+    {6, 0, 0, 0},
+
+    // level7 + default
+    {7, 0, 0, 0},
+
+    // level1 + default
+    {8, 0, 0, 0},
+
+    // level8 + default
+    {9, 0, 0, 0},
+
+    // level9 + default
+    {10, 0, 0, 0},
+
+    // level10 + default
+    {1, 0, 0, 0},
```

Comment:
Looking at the benchmark I was surprised to see level 10 compression (more compression / spend more cpu), take longer than level 9, this explains why :D. Should be 10 instead of 1.

Reply:
sry, fixed, thanks.